{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1>About the GTEx Data</h1>\n",
    "\n",
    "The scripts shown here can be used to pull gene expression data for a user-specified list of genes from the <b>Genotype-Tissue Expression (GTEx)</b> project data. This data set consists of gene expression data from over 50 primary tissue types. Each tissue has samples from multiple individuals and (usually) both sexes, collected from donors across the United States.\n",
    "\n",
    "The publicly-available data files can be downloaded from https://www.gtexportal.org/home/datasets after creating a free account. Additional raw and protected data is available through dbGaP.\n",
    "\n",
    "I use the v4 release because it is no longer under embargo as of January 2015.\n",
    "\n",
    "<h1>Analysis</h1>\n",
    "\n",
    "The workflow below can be used to calculate average expression levels across males and females separately for each tissue subtype in the data. This was used in <a href=\"https://www.ncbi.nlm.nih.gov/pubmed/26494842\">Slavney et al. 2016 (MBE)</a> to generate the data used in figure 2, but using a different gene list as input.\n",
    "\n",
    "Unless otherwise noted, all code was written by A. Slavney (2015)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h1>Input Data</h1>\n",
    "<h2>Gene expression values (RPKM)</h2>\n",
    "\n",
    "A matrix of gene expression values in reads per billion. Cols = genes, rows = samples.<br>\n",
    "File: GTEx_Analysis_V4_RNA-seq_RNA-SeQCv1.1.8_gene_rpkm.gct\n",
    "\n",
    "<h2>Deidentified subject data</h2>\n",
    "\n",
    "A file with sex, age, and cause of death data for each tissue sample.<br>\n",
    "File: GTEx_Data_V4_Annotations_SubjectPhenotypes_DS.txt\n",
    "\n",
    "<h2>Deidentified sample attribute data</h2>\n",
    "\n",
    "A file with detailed sample collection and processing information for each tissue sample.<br>\n",
    "File: GTEx_Data_V4_Annotations_SampleAttributesDS.txt\n",
    "\n",
    "<h2>Query gene list</h2>\n",
    "\n",
    "A file containing a list of Ensembl gene and transcript IDs for which we will obtain expression values.<br>\n",
    "File: Example_genes.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Import modules, define some functions\n",
    "\n",
    "import numpy, itertools, re, math\n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "\n",
    "def filter_expr(tissue_dict, sample_ids, gene_list):\n",
    "    ''' Get log2(RPKM) values for all individuals in each tissue '''\n",
    "    expr = defaultdict(list)\n",
    "    zeros = []\n",
    "    for k,v in tissue_dict.items(): # k = tissue, v = sample IDs\n",
    "        inds = []\n",
    "        # Make list of column positions in RPKM file for every sample ID of the current sex for this tissue\n",
    "        for j in v:\n",
    "            try:\n",
    "                ind = sample_ids.index(j)\n",
    "                if sample_ids[ind] != j:\n",
    "                    print('Sample ID mismatch:',j)\n",
    "                    raise KeyboardInterrupt\n",
    "            # Many of these samples are not in the RPKM file\n",
    "            except ValueError:\n",
    "                continue\n",
    "            inds.append(ind)\n",
    "        # Get expression values for every gene in this tissue\n",
    "        for g in gene_list:\n",
    "            # Expr vals of gene g for all individuals in tissue k\n",
    "            vals = [float(g[n]) for n in inds]\n",
    "            # Record log2 of non-zero RPKM values for all individual in tissue k\n",
    "            vals = [math.log(i,2) for i in vals if i>0]\n",
    "            if vals!=[]:\n",
    "                # Add entry to expr: k = gene ID, v = [tissue, [expr. vals. from all inds.]]\n",
    "                expr[g[0].split('.')[0]].append([k,vals])\n",
    "            else:\n",
    "                zeros.append(g[0].split('.')[0])\n",
    "    return([expr,zeros])\n",
    "\n",
    "\n",
    "def bootstrap(data, n):\n",
    "    ''' Generate 1000 bootstrap samples of some values and calculate the mean across the samples '''\n",
    "        \n",
    "    out = []\n",
    "    L = len(data)\n",
    "    # Generate n lists of L random integers, value 0-L, with replacement\n",
    "    samples = numpy.random.randint(0,L,(n,L))\n",
    "    for s in samples:\n",
    "        inds = [data[i] for i in s]\n",
    "        out.append(numpy.mean(inds))\n",
    "    return(numpy.mean(out))\n",
    "\n",
    "\n",
    "def avg_tissue_expr(expr, tissue_list, N, low):\n",
    "    \n",
    "    ''' Get the average expression value across individuals for each gene in each tissue,\n",
    "        after removing expression values below a chosen cutoff value (e.g. 1 RPKM)'''\n",
    "    \n",
    "    out_df = pd.DataFrame(columns=[i[1] for i in tissue_list])\n",
    "    out_df\n",
    "    \n",
    "    for k,v in expr.items():\n",
    "        gene = k\n",
    "        # Initialize row for this gene with all NA values\n",
    "        line = ['NA' for t in tissue_list]\n",
    "        # List of tissues w/ any expr. data for this gene\n",
    "        g_tissues = [j[0] for j in v]\n",
    "        if len(g_tissues)==0:\n",
    "            print('no data:', gene)\n",
    "        for t in tissue_list: \n",
    "            if t[1] in g_tissues:\n",
    "                ind = g_tissues.index(t[1])\n",
    "                good_expr = [i for i in v[ind][1] if i>=low]\n",
    "                if good_expr == []:\n",
    "                    line[t[0]] = 'NA'\n",
    "                else:\n",
    "                    # Add the bootstrapped (across individuals) avg. expr. val. of this gene to the line in the appropriate column (tissue)\n",
    "                    line[t[0]] = bootstrap(good_expr, N)\n",
    "        out_df.loc[gene] = line\n",
    "    return(out_df)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Getting Sex-Specific Data</h2>\n",
    "\n",
    "If we want to analyze the male and female samples separately, we need to pull the sex information for each sample from the phenotype file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "74 female subjects,  138 male subjects\n"
     ]
    }
   ],
   "source": [
    "## For each sex, count the number of individuals with data for at least one tissue type\n",
    "f_ids, m_ids =[],[]\n",
    "with open('GTEx_Data_V4_Annotations_SubjectPhenotypes_DS.txt') as PHENS:\n",
    "    for i in PHENS:\n",
    "        # Skip comment lines\n",
    "        if re.match('G+',i):\n",
    "            i = i.strip('\\n').split('\\t')\n",
    "            if i[1] == '2':\n",
    "                f_ids.append(i[0])\n",
    "            elif i[1] == '1':\n",
    "                m_ids.append(i[0])\n",
    "print(len(f_ids), 'female subjects, ', len(m_ids), 'male subjects')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Splitting Data by Tissue</h2>\n",
    "\n",
    "We also want to look at expression within each tissue subtype, so this bit of code sorts the sample IDs in each sex according to tissue."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## For each sex, make a dictionary of sample IDs (each sample = 1 tissue from 1 ind.) for each tissue type\n",
    "f_tissue_samples = defaultdict(list)\n",
    "m_tissue_samples = defaultdict(list)\n",
    "with open('GTEx_Data_V4_Annotations_SampleAttributesDS.txt') as ATTR:\n",
    "    for i in ATTR:\n",
    "        # Skip comment lines\n",
    "        if re.match('G+',i):\n",
    "            i = i.strip('\\n').split('\\t')\n",
    "            # Only interested in Illumina TrueSeq data\n",
    "            if i[13].strip() == 'TrueSeq.v1':\n",
    "                samp_id = i[0].strip()\n",
    "                subj_id = '-'.join(samp_id.split('-')[0:2])\n",
    "                # Use specific (vs. broad) tissue type\n",
    "                tissue = i[6]\n",
    "                # Skip cell line samples\n",
    "                if re.match('Cells -+',i[6]):\n",
    "                    continue\n",
    "                # Add sample IDs to either female or male list\n",
    "                if subj_id in f_ids:\n",
    "                    f_tissue_samples[tissue].append(samp_id)\n",
    "                elif subj_id in m_ids:\n",
    "                    m_tissue_samples[tissue].append(samp_id)\n",
    "                else:\n",
    "                    continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Pulling Expression Data for Query Genes</h2>\n",
    "\n",
    "Next, we will extract expression values for the genes listed in Example_genes.txt from the RPKM file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query genes: 49\n",
      "Query genes with expr. data: 47\n",
      "Female, male non-zero genes: 47 47\n"
     ]
    }
   ],
   "source": [
    "## Read in list of genes\n",
    "q_genes = []\n",
    "with open('Example_genes.txt') as GENES:\n",
    "    for i in GENES:\n",
    "        i = i.strip().split(',')\n",
    "        q_genes.append(i[0])\n",
    "        \n",
    "# Only keep unique gene IDs\n",
    "q_genes = list(set(q_genes))\n",
    "print('Query genes:',len(q_genes))\n",
    "    \n",
    "## Read in the RPKM file (cols = samples, rows = genes)         \n",
    "EXPR = open('GTEx_Analysis_2014-01-17_RNA-seq_RNA-SeQCv1.1.8_gene_rpkm.gct')\n",
    "# Skip comment lines\n",
    "rpkm = EXPR.read().split('\\n')[2:]\n",
    "EXPR.close()\n",
    "Sample_ids = rpkm[0].split('\\t')\n",
    "\n",
    "## Retrieve the expression data for query genes\n",
    "gene_data = []\n",
    "for i in rpkm:\n",
    "    i = i.split('\\t')\n",
    "    if i[0].split('.')[0] in q_genes:\n",
    "        gene_data.append(i)\n",
    "print('Query genes with expr. data:', len(gene_data))\n",
    "\n",
    "## Make dictionaries with keys = tissue, values = list of usable expr vals in this tissue for each gene\n",
    "\n",
    "f_expr = filter_expr(f_tissue_samples, Sample_ids, gene_data)\n",
    "m_expr = filter_expr(m_tissue_samples, Sample_ids, gene_data)\n",
    "\n",
    "print('Female, male non-zero genes:', len(f_expr[0]), len(m_expr[0]))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h2>Calculate Average Per Gene Per Tissue Expression Values</h2>\n",
    "\n",
    "Lastly, we'll calculate the bootstrap average log2 expression value of each gene in each tissue for females and males, and store them as pandas dataframes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Adipose - Subcutaneous</th>\n",
       "      <th>Adipose - Visceral (Omentum)</th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Artery - Aorta</th>\n",
       "      <th>Artery - Coronary</th>\n",
       "      <th>Artery - Tibial</th>\n",
       "      <th>Bladder</th>\n",
       "      <th>Brain - Amygdala</th>\n",
       "      <th>Brain - Anterior cingulate cortex (BA24)</th>\n",
       "      <th>Brain - Caudate (basal ganglia)</th>\n",
       "      <th>...</th>\n",
       "      <th>Pituitary</th>\n",
       "      <th>Skin - Not Sun Exposed (Suprapubic)</th>\n",
       "      <th>Skin - Sun Exposed (Lower leg)</th>\n",
       "      <th>Small Intestine - Terminal Ileum</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Stomach</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Uterus</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Whole Blood</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000134375</th>\n",
       "      <td>3.65648</td>\n",
       "      <td>3.61745</td>\n",
       "      <td>4.31073</td>\n",
       "      <td>3.49193</td>\n",
       "      <td>3.55371</td>\n",
       "      <td>3.22121</td>\n",
       "      <td>3.36649</td>\n",
       "      <td>3.45979</td>\n",
       "      <td>3.81704</td>\n",
       "      <td>3.43346</td>\n",
       "      <td>...</td>\n",
       "      <td>3.65944</td>\n",
       "      <td>3.09673</td>\n",
       "      <td>3.08517</td>\n",
       "      <td>3.44311</td>\n",
       "      <td>3.70013</td>\n",
       "      <td>3.50056</td>\n",
       "      <td>3.62664</td>\n",
       "      <td>3.37235</td>\n",
       "      <td>3.2246</td>\n",
       "      <td>1.4611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176393</th>\n",
       "      <td>4.27664</td>\n",
       "      <td>4.15954</td>\n",
       "      <td>4.6621</td>\n",
       "      <td>3.8653</td>\n",
       "      <td>3.84679</td>\n",
       "      <td>3.39429</td>\n",
       "      <td>4.22995</td>\n",
       "      <td>2.59654</td>\n",
       "      <td>2.5687</td>\n",
       "      <td>2.65992</td>\n",
       "      <td>...</td>\n",
       "      <td>3.80299</td>\n",
       "      <td>4.65408</td>\n",
       "      <td>4.63783</td>\n",
       "      <td>4.72384</td>\n",
       "      <td>5.00541</td>\n",
       "      <td>4.24201</td>\n",
       "      <td>4.83398</td>\n",
       "      <td>4.5147</td>\n",
       "      <td>4.5886</td>\n",
       "      <td>4.28085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000163435</th>\n",
       "      <td>2.02319</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>4.5627</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>1.52047</td>\n",
       "      <td>2.33107</td>\n",
       "      <td>2.14845</td>\n",
       "      <td>3.85308</td>\n",
       "      <td>NA</td>\n",
       "      <td>5.42013</td>\n",
       "      <td>1.3145</td>\n",
       "      <td>1.43154</td>\n",
       "      <td>5.28378</td>\n",
       "      <td>1.19254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000170075</th>\n",
       "      <td>0.818561</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>1.54383</td>\n",
       "      <td>NA</td>\n",
       "      <td>5.16764</td>\n",
       "      <td>4.25626</td>\n",
       "      <td>5.41518</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.0219945</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.302602</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143862</th>\n",
       "      <td>4.69373</td>\n",
       "      <td>4.63418</td>\n",
       "      <td>4.77154</td>\n",
       "      <td>4.75217</td>\n",
       "      <td>4.70712</td>\n",
       "      <td>4.57441</td>\n",
       "      <td>4.36466</td>\n",
       "      <td>5.12101</td>\n",
       "      <td>5.3832</td>\n",
       "      <td>5.07644</td>\n",
       "      <td>...</td>\n",
       "      <td>4.58491</td>\n",
       "      <td>4.62302</td>\n",
       "      <td>4.58252</td>\n",
       "      <td>4.17222</td>\n",
       "      <td>4.922</td>\n",
       "      <td>4.29906</td>\n",
       "      <td>4.6502</td>\n",
       "      <td>4.77608</td>\n",
       "      <td>4.62941</td>\n",
       "      <td>5.33385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143851</th>\n",
       "      <td>0.511803</td>\n",
       "      <td>0.678116</td>\n",
       "      <td>1.9702</td>\n",
       "      <td>0.542345</td>\n",
       "      <td>0.496112</td>\n",
       "      <td>0.245254</td>\n",
       "      <td>0.693856</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2.37271</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.0231334</td>\n",
       "      <td>0.468121</td>\n",
       "      <td>1.93301</td>\n",
       "      <td>3.54982</td>\n",
       "      <td>0.948228</td>\n",
       "      <td>0.74489</td>\n",
       "      <td>0.197802</td>\n",
       "      <td>0.533359</td>\n",
       "      <td>3.39633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000133067</th>\n",
       "      <td>0.628313</td>\n",
       "      <td>0.48932</td>\n",
       "      <td>NA</td>\n",
       "      <td>5.46013</td>\n",
       "      <td>5.28089</td>\n",
       "      <td>3.25426</td>\n",
       "      <td>1.0898</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2.8361</td>\n",
       "      <td>...</td>\n",
       "      <td>2.97291</td>\n",
       "      <td>1.19268</td>\n",
       "      <td>1.13026</td>\n",
       "      <td>NA</td>\n",
       "      <td>2.29211</td>\n",
       "      <td>0.795011</td>\n",
       "      <td>1.94999</td>\n",
       "      <td>1.03992</td>\n",
       "      <td>1.5764</td>\n",
       "      <td>1.84196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000077152</th>\n",
       "      <td>0.646395</td>\n",
       "      <td>0.44472</td>\n",
       "      <td>1.9124</td>\n",
       "      <td>0.547754</td>\n",
       "      <td>0.596359</td>\n",
       "      <td>0.59862</td>\n",
       "      <td>0.794656</td>\n",
       "      <td>2.3667</td>\n",
       "      <td>2.6625</td>\n",
       "      <td>2.69583</td>\n",
       "      <td>...</td>\n",
       "      <td>1.14419</td>\n",
       "      <td>1.50964</td>\n",
       "      <td>1.02486</td>\n",
       "      <td>1.8562</td>\n",
       "      <td>2.56812</td>\n",
       "      <td>1.6756</td>\n",
       "      <td>0.55965</td>\n",
       "      <td>1.10764</td>\n",
       "      <td>1.8132</td>\n",
       "      <td>1.14606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000077157</th>\n",
       "      <td>3.62044</td>\n",
       "      <td>3.05631</td>\n",
       "      <td>1.55519</td>\n",
       "      <td>6.23894</td>\n",
       "      <td>5.98442</td>\n",
       "      <td>6.4065</td>\n",
       "      <td>5.59927</td>\n",
       "      <td>2.60475</td>\n",
       "      <td>2.89998</td>\n",
       "      <td>2.60772</td>\n",
       "      <td>...</td>\n",
       "      <td>2.08124</td>\n",
       "      <td>2.74533</td>\n",
       "      <td>2.77191</td>\n",
       "      <td>4.54333</td>\n",
       "      <td>3.36492</td>\n",
       "      <td>3.70437</td>\n",
       "      <td>2.89011</td>\n",
       "      <td>5.99601</td>\n",
       "      <td>3.57969</td>\n",
       "      <td>1.39615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143858</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.446559</td>\n",
       "      <td>0.551855</td>\n",
       "      <td>0.260708</td>\n",
       "      <td>0.326138</td>\n",
       "      <td>0.0230013</td>\n",
       "      <td>NA</td>\n",
       "      <td>1.69117</td>\n",
       "      <td>2.48696</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.217917</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.505429</td>\n",
       "      <td>0.0463427</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000117139</th>\n",
       "      <td>2.65719</td>\n",
       "      <td>2.42641</td>\n",
       "      <td>2.57474</td>\n",
       "      <td>2.27957</td>\n",
       "      <td>2.27</td>\n",
       "      <td>2.44652</td>\n",
       "      <td>3.06843</td>\n",
       "      <td>1.66693</td>\n",
       "      <td>1.89982</td>\n",
       "      <td>1.9617</td>\n",
       "      <td>...</td>\n",
       "      <td>2.44654</td>\n",
       "      <td>3.93704</td>\n",
       "      <td>4.02039</td>\n",
       "      <td>2.45089</td>\n",
       "      <td>2.4612</td>\n",
       "      <td>2.55735</td>\n",
       "      <td>2.89107</td>\n",
       "      <td>3.20333</td>\n",
       "      <td>3.41505</td>\n",
       "      <td>2.31929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000183155</th>\n",
       "      <td>1.35333</td>\n",
       "      <td>1.41115</td>\n",
       "      <td>1.83072</td>\n",
       "      <td>1.70195</td>\n",
       "      <td>1.62859</td>\n",
       "      <td>1.16979</td>\n",
       "      <td>1.41472</td>\n",
       "      <td>2.29706</td>\n",
       "      <td>2.66461</td>\n",
       "      <td>2.29211</td>\n",
       "      <td>...</td>\n",
       "      <td>1.89169</td>\n",
       "      <td>1.56279</td>\n",
       "      <td>1.62271</td>\n",
       "      <td>1.88281</td>\n",
       "      <td>2.32236</td>\n",
       "      <td>1.5053</td>\n",
       "      <td>1.50373</td>\n",
       "      <td>1.49401</td>\n",
       "      <td>1.82072</td>\n",
       "      <td>2.17832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000117153</th>\n",
       "      <td>3.29003</td>\n",
       "      <td>3.09508</td>\n",
       "      <td>3.65479</td>\n",
       "      <td>2.95609</td>\n",
       "      <td>3.05733</td>\n",
       "      <td>2.93756</td>\n",
       "      <td>3.51024</td>\n",
       "      <td>3.15544</td>\n",
       "      <td>3.48933</td>\n",
       "      <td>3.24217</td>\n",
       "      <td>...</td>\n",
       "      <td>3.49622</td>\n",
       "      <td>3.07216</td>\n",
       "      <td>3.10081</td>\n",
       "      <td>3.09656</td>\n",
       "      <td>3.58877</td>\n",
       "      <td>3.06129</td>\n",
       "      <td>3.54566</td>\n",
       "      <td>3.30785</td>\n",
       "      <td>3.26386</td>\n",
       "      <td>2.34111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000159346</th>\n",
       "      <td>5.41229</td>\n",
       "      <td>5.28535</td>\n",
       "      <td>5.32235</td>\n",
       "      <td>5.19054</td>\n",
       "      <td>5.18482</td>\n",
       "      <td>5.13986</td>\n",
       "      <td>5.19021</td>\n",
       "      <td>4.87003</td>\n",
       "      <td>4.92007</td>\n",
       "      <td>5.12293</td>\n",
       "      <td>...</td>\n",
       "      <td>5.38202</td>\n",
       "      <td>6.20102</td>\n",
       "      <td>6.34451</td>\n",
       "      <td>4.81667</td>\n",
       "      <td>5.78058</td>\n",
       "      <td>5.33294</td>\n",
       "      <td>5.10043</td>\n",
       "      <td>4.90679</td>\n",
       "      <td>5.61369</td>\n",
       "      <td>7.91366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000154813</th>\n",
       "      <td>2.82475</td>\n",
       "      <td>3.04276</td>\n",
       "      <td>2.88731</td>\n",
       "      <td>2.98036</td>\n",
       "      <td>3.17911</td>\n",
       "      <td>3.02675</td>\n",
       "      <td>2.71283</td>\n",
       "      <td>2.79006</td>\n",
       "      <td>2.95239</td>\n",
       "      <td>2.81248</td>\n",
       "      <td>...</td>\n",
       "      <td>3.05981</td>\n",
       "      <td>1.46559</td>\n",
       "      <td>1.38608</td>\n",
       "      <td>3.18105</td>\n",
       "      <td>2.37088</td>\n",
       "      <td>2.5747</td>\n",
       "      <td>2.49639</td>\n",
       "      <td>2.71787</td>\n",
       "      <td>2.08709</td>\n",
       "      <td>1.9748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000154814</th>\n",
       "      <td>0.945575</td>\n",
       "      <td>0.744163</td>\n",
       "      <td>0.720352</td>\n",
       "      <td>0.289937</td>\n",
       "      <td>0.319232</td>\n",
       "      <td>0.308709</td>\n",
       "      <td>0.631683</td>\n",
       "      <td>0.169604</td>\n",
       "      <td>0.594522</td>\n",
       "      <td>0.308458</td>\n",
       "      <td>...</td>\n",
       "      <td>0.619881</td>\n",
       "      <td>0.455002</td>\n",
       "      <td>0.541966</td>\n",
       "      <td>1.58708</td>\n",
       "      <td>0.898922</td>\n",
       "      <td>0.990517</td>\n",
       "      <td>1.12584</td>\n",
       "      <td>0.392865</td>\n",
       "      <td>0.485879</td>\n",
       "      <td>0.881942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000131378</th>\n",
       "      <td>4.65233</td>\n",
       "      <td>4.27894</td>\n",
       "      <td>1.81625</td>\n",
       "      <td>4.55124</td>\n",
       "      <td>3.2603</td>\n",
       "      <td>3.62351</td>\n",
       "      <td>2.71651</td>\n",
       "      <td>1.5477</td>\n",
       "      <td>2.60566</td>\n",
       "      <td>0.89378</td>\n",
       "      <td>...</td>\n",
       "      <td>1.17046</td>\n",
       "      <td>2.95367</td>\n",
       "      <td>2.90437</td>\n",
       "      <td>2.77485</td>\n",
       "      <td>4.45683</td>\n",
       "      <td>2.04226</td>\n",
       "      <td>4.86649</td>\n",
       "      <td>4.06292</td>\n",
       "      <td>2.50174</td>\n",
       "      <td>2.09365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000092345</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000154822</th>\n",
       "      <td>2.26061</td>\n",
       "      <td>2.05655</td>\n",
       "      <td>2.87578</td>\n",
       "      <td>1.11013</td>\n",
       "      <td>1.20753</td>\n",
       "      <td>1.42943</td>\n",
       "      <td>0.631904</td>\n",
       "      <td>1.11605</td>\n",
       "      <td>1.8177</td>\n",
       "      <td>1.95486</td>\n",
       "      <td>...</td>\n",
       "      <td>2.48204</td>\n",
       "      <td>0.455453</td>\n",
       "      <td>0.616747</td>\n",
       "      <td>2.12192</td>\n",
       "      <td>3.06644</td>\n",
       "      <td>1.54339</td>\n",
       "      <td>1.54216</td>\n",
       "      <td>1.66583</td>\n",
       "      <td>0.980061</td>\n",
       "      <td>2.85432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000162105</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.266838</td>\n",
       "      <td>2.04609</td>\n",
       "      <td>1.78302</td>\n",
       "      <td>1.37404</td>\n",
       "      <td>...</td>\n",
       "      <td>1.11463</td>\n",
       "      <td>0.45838</td>\n",
       "      <td>0.164165</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.499187</td>\n",
       "      <td>1.58025</td>\n",
       "      <td>0.210869</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172893</th>\n",
       "      <td>2.55151</td>\n",
       "      <td>2.90903</td>\n",
       "      <td>5.31883</td>\n",
       "      <td>2.03543</td>\n",
       "      <td>2.22186</td>\n",
       "      <td>1.58802</td>\n",
       "      <td>1.71909</td>\n",
       "      <td>3.24905</td>\n",
       "      <td>3.38383</td>\n",
       "      <td>2.88844</td>\n",
       "      <td>...</td>\n",
       "      <td>2.90939</td>\n",
       "      <td>4.53777</td>\n",
       "      <td>4.808</td>\n",
       "      <td>2.8555</td>\n",
       "      <td>2.4961</td>\n",
       "      <td>2.99687</td>\n",
       "      <td>2.53838</td>\n",
       "      <td>2.18248</td>\n",
       "      <td>3.11666</td>\n",
       "      <td>2.06699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172890</th>\n",
       "      <td>2.91138</td>\n",
       "      <td>2.95768</td>\n",
       "      <td>3.21444</td>\n",
       "      <td>2.96117</td>\n",
       "      <td>3.04576</td>\n",
       "      <td>2.70374</td>\n",
       "      <td>3.81261</td>\n",
       "      <td>1.98366</td>\n",
       "      <td>1.79913</td>\n",
       "      <td>2.00262</td>\n",
       "      <td>...</td>\n",
       "      <td>3.85676</td>\n",
       "      <td>3.21429</td>\n",
       "      <td>3.24994</td>\n",
       "      <td>3.96225</td>\n",
       "      <td>4.07231</td>\n",
       "      <td>3.25409</td>\n",
       "      <td>3.59791</td>\n",
       "      <td>3.28586</td>\n",
       "      <td>3.47377</td>\n",
       "      <td>3.66222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000244411</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.018888</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.876622</td>\n",
       "      <td>1.78675</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000241233</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.492169</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.897563</td>\n",
       "      <td>1.87655</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000254997</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2.14547</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.0704529</td>\n",
       "      <td>...</td>\n",
       "      <td>0.139128</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.58797</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.576533</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204572</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.989884</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>1.75618</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000104129</th>\n",
       "      <td>2.50742</td>\n",
       "      <td>2.44516</td>\n",
       "      <td>2.33609</td>\n",
       "      <td>2.27472</td>\n",
       "      <td>2.15674</td>\n",
       "      <td>2.07019</td>\n",
       "      <td>2.28066</td>\n",
       "      <td>1.49573</td>\n",
       "      <td>1.42653</td>\n",
       "      <td>1.45686</td>\n",
       "      <td>...</td>\n",
       "      <td>2.38122</td>\n",
       "      <td>2.83495</td>\n",
       "      <td>2.74993</td>\n",
       "      <td>2.47212</td>\n",
       "      <td>2.8483</td>\n",
       "      <td>2.27637</td>\n",
       "      <td>2.8629</td>\n",
       "      <td>2.79543</td>\n",
       "      <td>2.79603</td>\n",
       "      <td>1.49716</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000166140</th>\n",
       "      <td>3.01581</td>\n",
       "      <td>2.91407</td>\n",
       "      <td>3.41498</td>\n",
       "      <td>2.85732</td>\n",
       "      <td>2.94764</td>\n",
       "      <td>2.51395</td>\n",
       "      <td>3.53991</td>\n",
       "      <td>2.98173</td>\n",
       "      <td>3.0244</td>\n",
       "      <td>3.14084</td>\n",
       "      <td>...</td>\n",
       "      <td>3.45304</td>\n",
       "      <td>3.6992</td>\n",
       "      <td>3.57959</td>\n",
       "      <td>3.40889</td>\n",
       "      <td>3.96578</td>\n",
       "      <td>3.41243</td>\n",
       "      <td>3.52116</td>\n",
       "      <td>3.3422</td>\n",
       "      <td>3.60597</td>\n",
       "      <td>2.39425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000166143</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>5.62212</td>\n",
       "      <td>NA</td>\n",
       "      <td>1.01218</td>\n",
       "      <td>0.941811</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000166145</th>\n",
       "      <td>1.44656</td>\n",
       "      <td>0.810538</td>\n",
       "      <td>0.396488</td>\n",
       "      <td>0.569827</td>\n",
       "      <td>1.37829</td>\n",
       "      <td>1.20531</td>\n",
       "      <td>3.05281</td>\n",
       "      <td>0.0585574</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>4.68787</td>\n",
       "      <td>6.33186</td>\n",
       "      <td>6.46862</td>\n",
       "      <td>4.43419</td>\n",
       "      <td>2.25675</td>\n",
       "      <td>5.54162</td>\n",
       "      <td>4.49487</td>\n",
       "      <td>1.47003</td>\n",
       "      <td>5.24411</td>\n",
       "      <td>1.43407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000104140</th>\n",
       "      <td>2.07061</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>3.15835</td>\n",
       "      <td>0.508243</td>\n",
       "      <td>0.926579</td>\n",
       "      <td>0.144747</td>\n",
       "      <td>...</td>\n",
       "      <td>1.19721</td>\n",
       "      <td>6.6049</td>\n",
       "      <td>6.553</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>1.15133</td>\n",
       "      <td>0.65437</td>\n",
       "      <td>NA</td>\n",
       "      <td>5.35869</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000104142</th>\n",
       "      <td>2.83549</td>\n",
       "      <td>2.77611</td>\n",
       "      <td>3.3288</td>\n",
       "      <td>3.09598</td>\n",
       "      <td>2.93316</td>\n",
       "      <td>2.86129</td>\n",
       "      <td>3.11652</td>\n",
       "      <td>2.17404</td>\n",
       "      <td>2.47632</td>\n",
       "      <td>2.41412</td>\n",
       "      <td>...</td>\n",
       "      <td>3.13902</td>\n",
       "      <td>2.80799</td>\n",
       "      <td>2.89916</td>\n",
       "      <td>2.88719</td>\n",
       "      <td>3.46541</td>\n",
       "      <td>2.89145</td>\n",
       "      <td>2.92404</td>\n",
       "      <td>3.08021</td>\n",
       "      <td>2.337</td>\n",
       "      <td>3.31603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000128917</th>\n",
       "      <td>3.94675</td>\n",
       "      <td>2.7842</td>\n",
       "      <td>0.913819</td>\n",
       "      <td>1.66315</td>\n",
       "      <td>2.06668</td>\n",
       "      <td>2.19266</td>\n",
       "      <td>2.14387</td>\n",
       "      <td>0.746887</td>\n",
       "      <td>0.970844</td>\n",
       "      <td>1.30249</td>\n",
       "      <td>...</td>\n",
       "      <td>1.97563</td>\n",
       "      <td>2.04324</td>\n",
       "      <td>1.7713</td>\n",
       "      <td>1.6275</td>\n",
       "      <td>1.68091</td>\n",
       "      <td>1.22985</td>\n",
       "      <td>3.39358</td>\n",
       "      <td>3.07871</td>\n",
       "      <td>2.40612</td>\n",
       "      <td>0.245139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000128965</th>\n",
       "      <td>1.43372</td>\n",
       "      <td>3.11596</td>\n",
       "      <td>1.26646</td>\n",
       "      <td>1.57194</td>\n",
       "      <td>0.88063</td>\n",
       "      <td>1.0868</td>\n",
       "      <td>0.0451542</td>\n",
       "      <td>1.42341</td>\n",
       "      <td>1.73199</td>\n",
       "      <td>0.846143</td>\n",
       "      <td>...</td>\n",
       "      <td>0.89976</td>\n",
       "      <td>2.26399</td>\n",
       "      <td>1.98612</td>\n",
       "      <td>0.456954</td>\n",
       "      <td>0.759021</td>\n",
       "      <td>1.88638</td>\n",
       "      <td>2.72495</td>\n",
       "      <td>1.4001</td>\n",
       "      <td>2.94521</td>\n",
       "      <td>1.81057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000128908</th>\n",
       "      <td>2.98624</td>\n",
       "      <td>2.80957</td>\n",
       "      <td>2.43333</td>\n",
       "      <td>2.68052</td>\n",
       "      <td>2.83474</td>\n",
       "      <td>3.01007</td>\n",
       "      <td>3.05677</td>\n",
       "      <td>1.87464</td>\n",
       "      <td>1.87917</td>\n",
       "      <td>1.88794</td>\n",
       "      <td>...</td>\n",
       "      <td>2.66814</td>\n",
       "      <td>3.08321</td>\n",
       "      <td>3.18063</td>\n",
       "      <td>3.07319</td>\n",
       "      <td>2.96362</td>\n",
       "      <td>2.59645</td>\n",
       "      <td>3.05545</td>\n",
       "      <td>3.08376</td>\n",
       "      <td>2.90631</td>\n",
       "      <td>1.29867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178997</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000104147</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.719923</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.744222</td>\n",
       "      <td>0.279667</td>\n",
       "      <td>1.47217</td>\n",
       "      <td>0.921872</td>\n",
       "      <td>1.44412</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.869405</td>\n",
       "      <td>0.632863</td>\n",
       "      <td>0.51755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000137804</th>\n",
       "      <td>0.942024</td>\n",
       "      <td>0.396336</td>\n",
       "      <td>3.26691</td>\n",
       "      <td>0.697462</td>\n",
       "      <td>0.894323</td>\n",
       "      <td>0.579455</td>\n",
       "      <td>0.191329</td>\n",
       "      <td>0.207999</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>0.245845</td>\n",
       "      <td>2.25358</td>\n",
       "      <td>1.69159</td>\n",
       "      <td>1.60387</td>\n",
       "      <td>2.81197</td>\n",
       "      <td>2.44334</td>\n",
       "      <td>0.220994</td>\n",
       "      <td>0.696038</td>\n",
       "      <td>2.14658</td>\n",
       "      <td>1.93704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000137806</th>\n",
       "      <td>2.9845</td>\n",
       "      <td>2.87085</td>\n",
       "      <td>4.52277</td>\n",
       "      <td>2.55807</td>\n",
       "      <td>2.65493</td>\n",
       "      <td>2.49386</td>\n",
       "      <td>2.91267</td>\n",
       "      <td>3.37966</td>\n",
       "      <td>3.45532</td>\n",
       "      <td>3.44749</td>\n",
       "      <td>...</td>\n",
       "      <td>3.12443</td>\n",
       "      <td>2.48161</td>\n",
       "      <td>2.54139</td>\n",
       "      <td>2.8028</td>\n",
       "      <td>3.17538</td>\n",
       "      <td>2.98386</td>\n",
       "      <td>2.99537</td>\n",
       "      <td>2.76301</td>\n",
       "      <td>2.74185</td>\n",
       "      <td>2.34381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000137815</th>\n",
       "      <td>3.71205</td>\n",
       "      <td>3.38327</td>\n",
       "      <td>4.05812</td>\n",
       "      <td>3.51865</td>\n",
       "      <td>3.50389</td>\n",
       "      <td>3.99942</td>\n",
       "      <td>4.08192</td>\n",
       "      <td>3.65129</td>\n",
       "      <td>3.78828</td>\n",
       "      <td>3.62462</td>\n",
       "      <td>...</td>\n",
       "      <td>3.31748</td>\n",
       "      <td>3.81366</td>\n",
       "      <td>3.95373</td>\n",
       "      <td>3.46996</td>\n",
       "      <td>3.39824</td>\n",
       "      <td>3.35906</td>\n",
       "      <td>3.88323</td>\n",
       "      <td>3.96256</td>\n",
       "      <td>3.75136</td>\n",
       "      <td>2.7464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000137825</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2.52676</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.0334439</td>\n",
       "      <td>3.3395</td>\n",
       "      <td>3.60183</td>\n",
       "      <td>4.07391</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.196938</td>\n",
       "      <td>0.00739313</td>\n",
       "      <td>4.07509</td>\n",
       "      <td>NA</td>\n",
       "      <td>3.11477</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.450472</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000062524</th>\n",
       "      <td>0.0666462</td>\n",
       "      <td>0.905492</td>\n",
       "      <td>0.330619</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.521296</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.417838</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.364548</td>\n",
       "      <td>0.669727</td>\n",
       "      <td>...</td>\n",
       "      <td>0.719716</td>\n",
       "      <td>0.366168</td>\n",
       "      <td>0.107611</td>\n",
       "      <td>2.0448</td>\n",
       "      <td>2.34618</td>\n",
       "      <td>1.87948</td>\n",
       "      <td>0.448029</td>\n",
       "      <td>NA</td>\n",
       "      <td>0.718196</td>\n",
       "      <td>1.2375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000103932</th>\n",
       "      <td>2.39073</td>\n",
       "      <td>2.27967</td>\n",
       "      <td>2.64539</td>\n",
       "      <td>2.09206</td>\n",
       "      <td>2.22761</td>\n",
       "      <td>2.2212</td>\n",
       "      <td>2.79536</td>\n",
       "      <td>1.72757</td>\n",
       "      <td>1.88457</td>\n",
       "      <td>1.89585</td>\n",
       "      <td>...</td>\n",
       "      <td>2.52967</td>\n",
       "      <td>2.53057</td>\n",
       "      <td>2.65079</td>\n",
       "      <td>2.25835</td>\n",
       "      <td>2.63588</td>\n",
       "      <td>2.38176</td>\n",
       "      <td>2.55573</td>\n",
       "      <td>2.59679</td>\n",
       "      <td>2.37503</td>\n",
       "      <td>0.88045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000092445</th>\n",
       "      <td>3.12115</td>\n",
       "      <td>3.59986</td>\n",
       "      <td>2.12127</td>\n",
       "      <td>0.869288</td>\n",
       "      <td>1.35223</td>\n",
       "      <td>0.69467</td>\n",
       "      <td>0.77176</td>\n",
       "      <td>4.47074</td>\n",
       "      <td>4.6572</td>\n",
       "      <td>3.81455</td>\n",
       "      <td>...</td>\n",
       "      <td>0.128036</td>\n",
       "      <td>3.4527</td>\n",
       "      <td>3.4561</td>\n",
       "      <td>0.730012</td>\n",
       "      <td>2.88378</td>\n",
       "      <td>1.65992</td>\n",
       "      <td>0.356994</td>\n",
       "      <td>2.1038</td>\n",
       "      <td>2.22749</td>\n",
       "      <td>0.327822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000174197</th>\n",
       "      <td>1.58362</td>\n",
       "      <td>1.48959</td>\n",
       "      <td>1.73121</td>\n",
       "      <td>1.65305</td>\n",
       "      <td>1.68554</td>\n",
       "      <td>1.81607</td>\n",
       "      <td>1.94333</td>\n",
       "      <td>0.966786</td>\n",
       "      <td>0.971815</td>\n",
       "      <td>1.16654</td>\n",
       "      <td>...</td>\n",
       "      <td>2.05311</td>\n",
       "      <td>1.48265</td>\n",
       "      <td>1.52422</td>\n",
       "      <td>1.60341</td>\n",
       "      <td>1.74348</td>\n",
       "      <td>1.37497</td>\n",
       "      <td>2.08772</td>\n",
       "      <td>2.15443</td>\n",
       "      <td>1.72328</td>\n",
       "      <td>0.621713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000137802</th>\n",
       "      <td>2.22053</td>\n",
       "      <td>2.15429</td>\n",
       "      <td>2.98441</td>\n",
       "      <td>2.6836</td>\n",
       "      <td>2.68298</td>\n",
       "      <td>2.65437</td>\n",
       "      <td>3.21947</td>\n",
       "      <td>2.38806</td>\n",
       "      <td>2.60612</td>\n",
       "      <td>2.6924</td>\n",
       "      <td>...</td>\n",
       "      <td>3.94509</td>\n",
       "      <td>3.94752</td>\n",
       "      <td>3.92704</td>\n",
       "      <td>2.77057</td>\n",
       "      <td>2.66074</td>\n",
       "      <td>2.75563</td>\n",
       "      <td>3.66872</td>\n",
       "      <td>3.12518</td>\n",
       "      <td>3.22602</td>\n",
       "      <td>0.805561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000207766</th>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Adipose - Subcutaneous Adipose - Visceral (Omentum)  \\\n",
       "ENSG00000134375                3.65648                      3.61745   \n",
       "ENSG00000176393                4.27664                      4.15954   \n",
       "ENSG00000163435                2.02319                           NA   \n",
       "ENSG00000170075               0.818561                           NA   \n",
       "ENSG00000143862                4.69373                      4.63418   \n",
       "ENSG00000143851               0.511803                     0.678116   \n",
       "ENSG00000133067               0.628313                      0.48932   \n",
       "ENSG00000077152               0.646395                      0.44472   \n",
       "ENSG00000077157                3.62044                      3.05631   \n",
       "ENSG00000143858                     NA                           NA   \n",
       "ENSG00000117139                2.65719                      2.42641   \n",
       "ENSG00000183155                1.35333                      1.41115   \n",
       "ENSG00000117153                3.29003                      3.09508   \n",
       "ENSG00000159346                5.41229                      5.28535   \n",
       "ENSG00000154813                2.82475                      3.04276   \n",
       "ENSG00000154814               0.945575                     0.744163   \n",
       "ENSG00000131378                4.65233                      4.27894   \n",
       "ENSG00000092345                     NA                           NA   \n",
       "ENSG00000154822                2.26061                      2.05655   \n",
       "ENSG00000162105                     NA                           NA   \n",
       "ENSG00000172893                2.55151                      2.90903   \n",
       "ENSG00000172890                2.91138                      2.95768   \n",
       "ENSG00000244411                     NA                           NA   \n",
       "ENSG00000241233                     NA                           NA   \n",
       "ENSG00000254997                     NA                           NA   \n",
       "ENSG00000204572                     NA                           NA   \n",
       "ENSG00000104129                2.50742                      2.44516   \n",
       "ENSG00000166140                3.01581                      2.91407   \n",
       "ENSG00000166143                     NA                           NA   \n",
       "ENSG00000166145                1.44656                     0.810538   \n",
       "ENSG00000104140                2.07061                           NA   \n",
       "ENSG00000104142                2.83549                      2.77611   \n",
       "ENSG00000128917                3.94675                       2.7842   \n",
       "ENSG00000128965                1.43372                      3.11596   \n",
       "ENSG00000128908                2.98624                      2.80957   \n",
       "ENSG00000178997                     NA                           NA   \n",
       "ENSG00000104147                     NA                           NA   \n",
       "ENSG00000137804               0.942024                     0.396336   \n",
       "ENSG00000137806                 2.9845                      2.87085   \n",
       "ENSG00000137815                3.71205                      3.38327   \n",
       "ENSG00000137825                     NA                           NA   \n",
       "ENSG00000062524              0.0666462                     0.905492   \n",
       "ENSG00000103932                2.39073                      2.27967   \n",
       "ENSG00000092445                3.12115                      3.59986   \n",
       "ENSG00000174197                1.58362                      1.48959   \n",
       "ENSG00000137802                2.22053                      2.15429   \n",
       "ENSG00000207766                     NA                           NA   \n",
       "\n",
       "                Adrenal Gland Artery - Aorta Artery - Coronary  \\\n",
       "ENSG00000134375       4.31073        3.49193           3.55371   \n",
       "ENSG00000176393        4.6621         3.8653           3.84679   \n",
       "ENSG00000163435            NA             NA                NA   \n",
       "ENSG00000170075            NA             NA                NA   \n",
       "ENSG00000143862       4.77154        4.75217           4.70712   \n",
       "ENSG00000143851        1.9702       0.542345          0.496112   \n",
       "ENSG00000133067            NA        5.46013           5.28089   \n",
       "ENSG00000077152        1.9124       0.547754          0.596359   \n",
       "ENSG00000077157       1.55519        6.23894           5.98442   \n",
       "ENSG00000143858      0.446559       0.551855          0.260708   \n",
       "ENSG00000117139       2.57474        2.27957              2.27   \n",
       "ENSG00000183155       1.83072        1.70195           1.62859   \n",
       "ENSG00000117153       3.65479        2.95609           3.05733   \n",
       "ENSG00000159346       5.32235        5.19054           5.18482   \n",
       "ENSG00000154813       2.88731        2.98036           3.17911   \n",
       "ENSG00000154814      0.720352       0.289937          0.319232   \n",
       "ENSG00000131378       1.81625        4.55124            3.2603   \n",
       "ENSG00000092345            NA             NA                NA   \n",
       "ENSG00000154822       2.87578        1.11013           1.20753   \n",
       "ENSG00000162105            NA             NA                NA   \n",
       "ENSG00000172893       5.31883        2.03543           2.22186   \n",
       "ENSG00000172890       3.21444        2.96117           3.04576   \n",
       "ENSG00000244411            NA             NA                NA   \n",
       "ENSG00000241233            NA             NA                NA   \n",
       "ENSG00000254997            NA             NA                NA   \n",
       "ENSG00000204572            NA             NA                NA   \n",
       "ENSG00000104129       2.33609        2.27472           2.15674   \n",
       "ENSG00000166140       3.41498        2.85732           2.94764   \n",
       "ENSG00000166143            NA             NA                NA   \n",
       "ENSG00000166145      0.396488       0.569827           1.37829   \n",
       "ENSG00000104140            NA             NA                NA   \n",
       "ENSG00000104142        3.3288        3.09598           2.93316   \n",
       "ENSG00000128917      0.913819        1.66315           2.06668   \n",
       "ENSG00000128965       1.26646        1.57194           0.88063   \n",
       "ENSG00000128908       2.43333        2.68052           2.83474   \n",
       "ENSG00000178997            NA             NA                NA   \n",
       "ENSG00000104147      0.719923             NA                NA   \n",
       "ENSG00000137804       3.26691       0.697462          0.894323   \n",
       "ENSG00000137806       4.52277        2.55807           2.65493   \n",
       "ENSG00000137815       4.05812        3.51865           3.50389   \n",
       "ENSG00000137825       2.52676             NA                NA   \n",
       "ENSG00000062524      0.330619             NA          0.521296   \n",
       "ENSG00000103932       2.64539        2.09206           2.22761   \n",
       "ENSG00000092445       2.12127       0.869288           1.35223   \n",
       "ENSG00000174197       1.73121        1.65305           1.68554   \n",
       "ENSG00000137802       2.98441         2.6836           2.68298   \n",
       "ENSG00000207766            NA             NA                NA   \n",
       "\n",
       "                Artery - Tibial    Bladder Brain - Amygdala  \\\n",
       "ENSG00000134375         3.22121    3.36649          3.45979   \n",
       "ENSG00000176393         3.39429    4.22995          2.59654   \n",
       "ENSG00000163435              NA     4.5627               NA   \n",
       "ENSG00000170075         1.54383         NA          5.16764   \n",
       "ENSG00000143862         4.57441    4.36466          5.12101   \n",
       "ENSG00000143851        0.245254   0.693856               NA   \n",
       "ENSG00000133067         3.25426     1.0898               NA   \n",
       "ENSG00000077152         0.59862   0.794656           2.3667   \n",
       "ENSG00000077157          6.4065    5.59927          2.60475   \n",
       "ENSG00000143858        0.326138  0.0230013               NA   \n",
       "ENSG00000117139         2.44652    3.06843          1.66693   \n",
       "ENSG00000183155         1.16979    1.41472          2.29706   \n",
       "ENSG00000117153         2.93756    3.51024          3.15544   \n",
       "ENSG00000159346         5.13986    5.19021          4.87003   \n",
       "ENSG00000154813         3.02675    2.71283          2.79006   \n",
       "ENSG00000154814        0.308709   0.631683         0.169604   \n",
       "ENSG00000131378         3.62351    2.71651           1.5477   \n",
       "ENSG00000092345              NA         NA               NA   \n",
       "ENSG00000154822         1.42943   0.631904          1.11605   \n",
       "ENSG00000162105              NA   0.266838          2.04609   \n",
       "ENSG00000172893         1.58802    1.71909          3.24905   \n",
       "ENSG00000172890         2.70374    3.81261          1.98366   \n",
       "ENSG00000244411              NA         NA         0.018888   \n",
       "ENSG00000241233              NA   0.492169               NA   \n",
       "ENSG00000254997              NA    2.14547               NA   \n",
       "ENSG00000204572              NA   0.989884               NA   \n",
       "ENSG00000104129         2.07019    2.28066          1.49573   \n",
       "ENSG00000166140         2.51395    3.53991          2.98173   \n",
       "ENSG00000166143              NA         NA               NA   \n",
       "ENSG00000166145         1.20531    3.05281        0.0585574   \n",
       "ENSG00000104140              NA    3.15835         0.508243   \n",
       "ENSG00000104142         2.86129    3.11652          2.17404   \n",
       "ENSG00000128917         2.19266    2.14387         0.746887   \n",
       "ENSG00000128965          1.0868  0.0451542          1.42341   \n",
       "ENSG00000128908         3.01007    3.05677          1.87464   \n",
       "ENSG00000178997              NA         NA               NA   \n",
       "ENSG00000104147              NA         NA               NA   \n",
       "ENSG00000137804        0.579455   0.191329         0.207999   \n",
       "ENSG00000137806         2.49386    2.91267          3.37966   \n",
       "ENSG00000137815         3.99942    4.08192          3.65129   \n",
       "ENSG00000137825              NA  0.0334439           3.3395   \n",
       "ENSG00000062524              NA   0.417838               NA   \n",
       "ENSG00000103932          2.2212    2.79536          1.72757   \n",
       "ENSG00000092445         0.69467    0.77176          4.47074   \n",
       "ENSG00000174197         1.81607    1.94333         0.966786   \n",
       "ENSG00000137802         2.65437    3.21947          2.38806   \n",
       "ENSG00000207766              NA         NA               NA   \n",
       "\n",
       "                Brain - Anterior cingulate cortex (BA24)  \\\n",
       "ENSG00000134375                                  3.81704   \n",
       "ENSG00000176393                                   2.5687   \n",
       "ENSG00000163435                                       NA   \n",
       "ENSG00000170075                                  4.25626   \n",
       "ENSG00000143862                                   5.3832   \n",
       "ENSG00000143851                                       NA   \n",
       "ENSG00000133067                                       NA   \n",
       "ENSG00000077152                                   2.6625   \n",
       "ENSG00000077157                                  2.89998   \n",
       "ENSG00000143858                                  1.69117   \n",
       "ENSG00000117139                                  1.89982   \n",
       "ENSG00000183155                                  2.66461   \n",
       "ENSG00000117153                                  3.48933   \n",
       "ENSG00000159346                                  4.92007   \n",
       "ENSG00000154813                                  2.95239   \n",
       "ENSG00000154814                                 0.594522   \n",
       "ENSG00000131378                                  2.60566   \n",
       "ENSG00000092345                                       NA   \n",
       "ENSG00000154822                                   1.8177   \n",
       "ENSG00000162105                                  1.78302   \n",
       "ENSG00000172893                                  3.38383   \n",
       "ENSG00000172890                                  1.79913   \n",
       "ENSG00000244411                                       NA   \n",
       "ENSG00000241233                                       NA   \n",
       "ENSG00000254997                                       NA   \n",
       "ENSG00000204572                                       NA   \n",
       "ENSG00000104129                                  1.42653   \n",
       "ENSG00000166140                                   3.0244   \n",
       "ENSG00000166143                                       NA   \n",
       "ENSG00000166145                                       NA   \n",
       "ENSG00000104140                                 0.926579   \n",
       "ENSG00000104142                                  2.47632   \n",
       "ENSG00000128917                                 0.970844   \n",
       "ENSG00000128965                                  1.73199   \n",
       "ENSG00000128908                                  1.87917   \n",
       "ENSG00000178997                                       NA   \n",
       "ENSG00000104147                                       NA   \n",
       "ENSG00000137804                                       NA   \n",
       "ENSG00000137806                                  3.45532   \n",
       "ENSG00000137815                                  3.78828   \n",
       "ENSG00000137825                                  3.60183   \n",
       "ENSG00000062524                                 0.364548   \n",
       "ENSG00000103932                                  1.88457   \n",
       "ENSG00000092445                                   4.6572   \n",
       "ENSG00000174197                                 0.971815   \n",
       "ENSG00000137802                                  2.60612   \n",
       "ENSG00000207766                                       NA   \n",
       "\n",
       "                Brain - Caudate (basal ganglia)     ...     Pituitary  \\\n",
       "ENSG00000134375                         3.43346     ...       3.65944   \n",
       "ENSG00000176393                         2.65992     ...       3.80299   \n",
       "ENSG00000163435                              NA     ...       1.52047   \n",
       "ENSG00000170075                         5.41518     ...            NA   \n",
       "ENSG00000143862                         5.07644     ...       4.58491   \n",
       "ENSG00000143851                         2.37271     ...            NA   \n",
       "ENSG00000133067                          2.8361     ...       2.97291   \n",
       "ENSG00000077152                         2.69583     ...       1.14419   \n",
       "ENSG00000077157                         2.60772     ...       2.08124   \n",
       "ENSG00000143858                         2.48696     ...            NA   \n",
       "ENSG00000117139                          1.9617     ...       2.44654   \n",
       "ENSG00000183155                         2.29211     ...       1.89169   \n",
       "ENSG00000117153                         3.24217     ...       3.49622   \n",
       "ENSG00000159346                         5.12293     ...       5.38202   \n",
       "ENSG00000154813                         2.81248     ...       3.05981   \n",
       "ENSG00000154814                        0.308458     ...      0.619881   \n",
       "ENSG00000131378                         0.89378     ...       1.17046   \n",
       "ENSG00000092345                              NA     ...            NA   \n",
       "ENSG00000154822                         1.95486     ...       2.48204   \n",
       "ENSG00000162105                         1.37404     ...       1.11463   \n",
       "ENSG00000172893                         2.88844     ...       2.90939   \n",
       "ENSG00000172890                         2.00262     ...       3.85676   \n",
       "ENSG00000244411                              NA     ...            NA   \n",
       "ENSG00000241233                              NA     ...            NA   \n",
       "ENSG00000254997                       0.0704529     ...      0.139128   \n",
       "ENSG00000204572                              NA     ...            NA   \n",
       "ENSG00000104129                         1.45686     ...       2.38122   \n",
       "ENSG00000166140                         3.14084     ...       3.45304   \n",
       "ENSG00000166143                              NA     ...            NA   \n",
       "ENSG00000166145                              NA     ...       4.68787   \n",
       "ENSG00000104140                        0.144747     ...       1.19721   \n",
       "ENSG00000104142                         2.41412     ...       3.13902   \n",
       "ENSG00000128917                         1.30249     ...       1.97563   \n",
       "ENSG00000128965                        0.846143     ...       0.89976   \n",
       "ENSG00000128908                         1.88794     ...       2.66814   \n",
       "ENSG00000178997                              NA     ...            NA   \n",
       "ENSG00000104147                              NA     ...            NA   \n",
       "ENSG00000137804                              NA     ...      0.245845   \n",
       "ENSG00000137806                         3.44749     ...       3.12443   \n",
       "ENSG00000137815                         3.62462     ...       3.31748   \n",
       "ENSG00000137825                         4.07391     ...            NA   \n",
       "ENSG00000062524                        0.669727     ...      0.719716   \n",
       "ENSG00000103932                         1.89585     ...       2.52967   \n",
       "ENSG00000092445                         3.81455     ...      0.128036   \n",
       "ENSG00000174197                         1.16654     ...       2.05311   \n",
       "ENSG00000137802                          2.6924     ...       3.94509   \n",
       "ENSG00000207766                              NA     ...            NA   \n",
       "\n",
       "                Skin - Not Sun Exposed (Suprapubic)  \\\n",
       "ENSG00000134375                             3.09673   \n",
       "ENSG00000176393                             4.65408   \n",
       "ENSG00000163435                             2.33107   \n",
       "ENSG00000170075                           0.0219945   \n",
       "ENSG00000143862                             4.62302   \n",
       "ENSG00000143851                           0.0231334   \n",
       "ENSG00000133067                             1.19268   \n",
       "ENSG00000077152                             1.50964   \n",
       "ENSG00000077157                             2.74533   \n",
       "ENSG00000143858                                  NA   \n",
       "ENSG00000117139                             3.93704   \n",
       "ENSG00000183155                             1.56279   \n",
       "ENSG00000117153                             3.07216   \n",
       "ENSG00000159346                             6.20102   \n",
       "ENSG00000154813                             1.46559   \n",
       "ENSG00000154814                            0.455002   \n",
       "ENSG00000131378                             2.95367   \n",
       "ENSG00000092345                                  NA   \n",
       "ENSG00000154822                            0.455453   \n",
       "ENSG00000162105                             0.45838   \n",
       "ENSG00000172893                             4.53777   \n",
       "ENSG00000172890                             3.21429   \n",
       "ENSG00000244411                            0.876622   \n",
       "ENSG00000241233                            0.897563   \n",
       "ENSG00000254997                                  NA   \n",
       "ENSG00000204572                                  NA   \n",
       "ENSG00000104129                             2.83495   \n",
       "ENSG00000166140                              3.6992   \n",
       "ENSG00000166143                                  NA   \n",
       "ENSG00000166145                             6.33186   \n",
       "ENSG00000104140                              6.6049   \n",
       "ENSG00000104142                             2.80799   \n",
       "ENSG00000128917                             2.04324   \n",
       "ENSG00000128965                             2.26399   \n",
       "ENSG00000128908                             3.08321   \n",
       "ENSG00000178997                                  NA   \n",
       "ENSG00000104147                            0.744222   \n",
       "ENSG00000137804                             2.25358   \n",
       "ENSG00000137806                             2.48161   \n",
       "ENSG00000137815                             3.81366   \n",
       "ENSG00000137825                            0.196938   \n",
       "ENSG00000062524                            0.366168   \n",
       "ENSG00000103932                             2.53057   \n",
       "ENSG00000092445                              3.4527   \n",
       "ENSG00000174197                             1.48265   \n",
       "ENSG00000137802                             3.94752   \n",
       "ENSG00000207766                                  NA   \n",
       "\n",
       "                Skin - Sun Exposed (Lower leg)  \\\n",
       "ENSG00000134375                        3.08517   \n",
       "ENSG00000176393                        4.63783   \n",
       "ENSG00000163435                        2.14845   \n",
       "ENSG00000170075                             NA   \n",
       "ENSG00000143862                        4.58252   \n",
       "ENSG00000143851                       0.468121   \n",
       "ENSG00000133067                        1.13026   \n",
       "ENSG00000077152                        1.02486   \n",
       "ENSG00000077157                        2.77191   \n",
       "ENSG00000143858                             NA   \n",
       "ENSG00000117139                        4.02039   \n",
       "ENSG00000183155                        1.62271   \n",
       "ENSG00000117153                        3.10081   \n",
       "ENSG00000159346                        6.34451   \n",
       "ENSG00000154813                        1.38608   \n",
       "ENSG00000154814                       0.541966   \n",
       "ENSG00000131378                        2.90437   \n",
       "ENSG00000092345                             NA   \n",
       "ENSG00000154822                       0.616747   \n",
       "ENSG00000162105                       0.164165   \n",
       "ENSG00000172893                          4.808   \n",
       "ENSG00000172890                        3.24994   \n",
       "ENSG00000244411                        1.78675   \n",
       "ENSG00000241233                        1.87655   \n",
       "ENSG00000254997                        0.58797   \n",
       "ENSG00000204572                        1.75618   \n",
       "ENSG00000104129                        2.74993   \n",
       "ENSG00000166140                        3.57959   \n",
       "ENSG00000166143                             NA   \n",
       "ENSG00000166145                        6.46862   \n",
       "ENSG00000104140                          6.553   \n",
       "ENSG00000104142                        2.89916   \n",
       "ENSG00000128917                         1.7713   \n",
       "ENSG00000128965                        1.98612   \n",
       "ENSG00000128908                        3.18063   \n",
       "ENSG00000178997                             NA   \n",
       "ENSG00000104147                       0.279667   \n",
       "ENSG00000137804                        1.69159   \n",
       "ENSG00000137806                        2.54139   \n",
       "ENSG00000137815                        3.95373   \n",
       "ENSG00000137825                     0.00739313   \n",
       "ENSG00000062524                       0.107611   \n",
       "ENSG00000103932                        2.65079   \n",
       "ENSG00000092445                         3.4561   \n",
       "ENSG00000174197                        1.52422   \n",
       "ENSG00000137802                        3.92704   \n",
       "ENSG00000207766                             NA   \n",
       "\n",
       "                Small Intestine - Terminal Ileum    Spleen   Stomach  \\\n",
       "ENSG00000134375                          3.44311   3.70013   3.50056   \n",
       "ENSG00000176393                          4.72384   5.00541   4.24201   \n",
       "ENSG00000163435                          3.85308        NA   5.42013   \n",
       "ENSG00000170075                               NA        NA        NA   \n",
       "ENSG00000143862                          4.17222     4.922   4.29906   \n",
       "ENSG00000143851                          1.93301   3.54982  0.948228   \n",
       "ENSG00000133067                               NA   2.29211  0.795011   \n",
       "ENSG00000077152                           1.8562   2.56812    1.6756   \n",
       "ENSG00000077157                          4.54333   3.36492   3.70437   \n",
       "ENSG00000143858                               NA        NA  0.217917   \n",
       "ENSG00000117139                          2.45089    2.4612   2.55735   \n",
       "ENSG00000183155                          1.88281   2.32236    1.5053   \n",
       "ENSG00000117153                          3.09656   3.58877   3.06129   \n",
       "ENSG00000159346                          4.81667   5.78058   5.33294   \n",
       "ENSG00000154813                          3.18105   2.37088    2.5747   \n",
       "ENSG00000154814                          1.58708  0.898922  0.990517   \n",
       "ENSG00000131378                          2.77485   4.45683   2.04226   \n",
       "ENSG00000092345                               NA        NA        NA   \n",
       "ENSG00000154822                          2.12192   3.06644   1.54339   \n",
       "ENSG00000162105                               NA        NA  0.499187   \n",
       "ENSG00000172893                           2.8555    2.4961   2.99687   \n",
       "ENSG00000172890                          3.96225   4.07231   3.25409   \n",
       "ENSG00000244411                               NA        NA        NA   \n",
       "ENSG00000241233                               NA        NA        NA   \n",
       "ENSG00000254997                               NA        NA        NA   \n",
       "ENSG00000204572                               NA        NA        NA   \n",
       "ENSG00000104129                          2.47212    2.8483   2.27637   \n",
       "ENSG00000166140                          3.40889   3.96578   3.41243   \n",
       "ENSG00000166143                          5.62212        NA   1.01218   \n",
       "ENSG00000166145                          4.43419   2.25675   5.54162   \n",
       "ENSG00000104140                               NA        NA   1.15133   \n",
       "ENSG00000104142                          2.88719   3.46541   2.89145   \n",
       "ENSG00000128917                           1.6275   1.68091   1.22985   \n",
       "ENSG00000128965                         0.456954  0.759021   1.88638   \n",
       "ENSG00000128908                          3.07319   2.96362   2.59645   \n",
       "ENSG00000178997                               NA        NA        NA   \n",
       "ENSG00000104147                          1.47217  0.921872   1.44412   \n",
       "ENSG00000137804                          1.60387   2.81197   2.44334   \n",
       "ENSG00000137806                           2.8028   3.17538   2.98386   \n",
       "ENSG00000137815                          3.46996   3.39824   3.35906   \n",
       "ENSG00000137825                          4.07509        NA   3.11477   \n",
       "ENSG00000062524                           2.0448   2.34618   1.87948   \n",
       "ENSG00000103932                          2.25835   2.63588   2.38176   \n",
       "ENSG00000092445                         0.730012   2.88378   1.65992   \n",
       "ENSG00000174197                          1.60341   1.74348   1.37497   \n",
       "ENSG00000137802                          2.77057   2.66074   2.75563   \n",
       "ENSG00000207766                               NA        NA        NA   \n",
       "\n",
       "                  Thyroid    Uterus     Vagina Whole Blood  \n",
       "ENSG00000134375   3.62664   3.37235     3.2246      1.4611  \n",
       "ENSG00000176393   4.83398    4.5147     4.5886     4.28085  \n",
       "ENSG00000163435    1.3145   1.43154    5.28378     1.19254  \n",
       "ENSG00000170075        NA        NA   0.302602          NA  \n",
       "ENSG00000143862    4.6502   4.77608    4.62941     5.33385  \n",
       "ENSG00000143851   0.74489  0.197802   0.533359     3.39633  \n",
       "ENSG00000133067   1.94999   1.03992     1.5764     1.84196  \n",
       "ENSG00000077152   0.55965   1.10764     1.8132     1.14606  \n",
       "ENSG00000077157   2.89011   5.99601    3.57969     1.39615  \n",
       "ENSG00000143858        NA  0.505429  0.0463427          NA  \n",
       "ENSG00000117139   2.89107   3.20333    3.41505     2.31929  \n",
       "ENSG00000183155   1.50373   1.49401    1.82072     2.17832  \n",
       "ENSG00000117153   3.54566   3.30785    3.26386     2.34111  \n",
       "ENSG00000159346   5.10043   4.90679    5.61369     7.91366  \n",
       "ENSG00000154813   2.49639   2.71787    2.08709      1.9748  \n",
       "ENSG00000154814   1.12584  0.392865   0.485879    0.881942  \n",
       "ENSG00000131378   4.86649   4.06292    2.50174     2.09365  \n",
       "ENSG00000092345        NA        NA         NA          NA  \n",
       "ENSG00000154822   1.54216   1.66583   0.980061     2.85432  \n",
       "ENSG00000162105   1.58025  0.210869         NA          NA  \n",
       "ENSG00000172893   2.53838   2.18248    3.11666     2.06699  \n",
       "ENSG00000172890   3.59791   3.28586    3.47377     3.66222  \n",
       "ENSG00000244411        NA        NA         NA          NA  \n",
       "ENSG00000241233        NA        NA         NA          NA  \n",
       "ENSG00000254997        NA        NA   0.576533          NA  \n",
       "ENSG00000204572        NA        NA         NA          NA  \n",
       "ENSG00000104129    2.8629   2.79543    2.79603     1.49716  \n",
       "ENSG00000166140   3.52116    3.3422    3.60597     2.39425  \n",
       "ENSG00000166143  0.941811        NA         NA          NA  \n",
       "ENSG00000166145   4.49487   1.47003    5.24411     1.43407  \n",
       "ENSG00000104140   0.65437        NA    5.35869          NA  \n",
       "ENSG00000104142   2.92404   3.08021      2.337     3.31603  \n",
       "ENSG00000128917   3.39358   3.07871    2.40612    0.245139  \n",
       "ENSG00000128965   2.72495    1.4001    2.94521     1.81057  \n",
       "ENSG00000128908   3.05545   3.08376    2.90631     1.29867  \n",
       "ENSG00000178997        NA        NA         NA          NA  \n",
       "ENSG00000104147        NA  0.869405   0.632863     0.51755  \n",
       "ENSG00000137804  0.220994  0.696038    2.14658     1.93704  \n",
       "ENSG00000137806   2.99537   2.76301    2.74185     2.34381  \n",
       "ENSG00000137815   3.88323   3.96256    3.75136      2.7464  \n",
       "ENSG00000137825        NA        NA   0.450472          NA  \n",
       "ENSG00000062524  0.448029        NA   0.718196      1.2375  \n",
       "ENSG00000103932   2.55573   2.59679    2.37503     0.88045  \n",
       "ENSG00000092445  0.356994    2.1038    2.22749    0.327822  \n",
       "ENSG00000174197   2.08772   2.15443    1.72328    0.621713  \n",
       "ENSG00000137802   3.66872   3.12518    3.22602    0.805561  \n",
       "ENSG00000207766        NA        NA         NA          NA  \n",
       "\n",
       "[47 rows x 49 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\" For each gene, for each tissue, filter and average individual log2(RPKM) values less than the tissue-specific cutoff. \"\"\"\n",
    "\n",
    "f_tissue_list = list(enumerate(sorted(f_tissue_samples.keys())))\n",
    "m_tissue_list = list(enumerate(sorted(m_tissue_samples.keys())))\n",
    "\n",
    "## Use a minumum value of 0 (corresponding to a log2(1 RPKM))\n",
    "\n",
    "Female = avg_tissue_expr(f_expr[0], f_tissue_list, 1000, 0)\n",
    "Male = avg_tissue_expr(m_expr[0], m_tissue_list, 1000, 0)\n",
    "\n",
    "## Display the Female table\n",
    "Female"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Now, we can (for example) compare the male and female expression values for a given gene in a given tissue."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4.771543193905943\n",
      "4.832115188945298\n",
      "-0.0605719950393544\n",
      "0.9588838687912008\n"
     ]
    }
   ],
   "source": [
    "# Female-male log2 fold change for gene ENSG00000143862 in Adrenal Gland:\n",
    "\n",
    "print(Female.get_value('ENSG00000143862','Adrenal Gland'))\n",
    "print(Male.get_value('ENSG00000143862','Adrenal Gland'))\n",
    "\n",
    "logFC = Female.get_value('ENSG00000143862','Adrenal Gland')-Male.get_value('ENSG00000143862','Adrenal Gland')\n",
    "print(logFC)\n",
    "\n",
    "# Converted to fold change:\n",
    "print(2**logFC)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
